rllib/core/rl_module/rl_module.py

```diff
@@ -79,6 +80,27 @@ class RLModule(abc.ABC):
     @OverrideToImplementCustomLogic_CallToSuperRecommended
     def __init__(self, config: Mapping[str, Any] = None) -> None:
         self.config = config or {}
+        self.setup()
+        self._input_specs_train = self.input_specs_train()
+        self._output_specs_train = self.output_specs_train()
+        self._input_specs_exploration = self.input_specs_exploration()
+        self._output_specs_exploration = self.output_specs_exploration()
+        self._input_specs_inference = self.input_specs_inference()
+        self._output_specs_inference = self.output_specs_inference()
```

Comment:
Since RLModule will be a nn.Module (for torch), these cannot be property. because nn.Module has overriden getattr and will do some wonky stuff with the property if the name is mis-spelled or something. I could have them as functions, but then in the check_specs I had to call `getattr(self, specs_name)()`  which looks not intuitive because of the second parentheses. Having the specs stored like this, is a compromise for all. I can access them via `getattr(self, spec_name)` and also torch.nn.Module won't do something un-predictable with it.

Reply:
More info on the `nn.Module` issue: https://github.com/pytorch/pytorch/issues/49726

Reply:
oh, this is that ``@property`` problem. ok, if users don't have to write this (completely internal), it's probably fine.
if we need to keep repeating these lines, let's sit down and think a bit more about the puzzle :)

Reply:
nope it is supposed to be internal.